include/vigra/adjacency_list_graph.hxx

```diff
@@ -487,6 +496,122 @@ namespace vigra{
 
         static const bool is_directed = false;
 
+    public:
+
+        void reserveMaxNodeId(const index_type mxid ){
+            if(nodeNum()==0 ||  mxid>maxNodeId())
+                nodes_.reserve(mxid+1);
+        }
+
+        void reserveEdges(const size_t size ){
+            if(size>nodeNum())
```

Comment:
A test for `nodeNum()` in a function `reserveEdges()` looks odd. Please explain.